pkg/bootkube/status.go

```diff
@@ -25,7 +25,8 @@ const (
 	doesNotExist = "DoesNotExist"
 )
 
-func WaitUntilPodsRunning(pods []string, timeout time.Duration, selfHostedEtcd bool) error {
+// TODO: Don't pass the etcdServiceIP
```

Comment:
I might be missing something, but I agree this is undesirable - but also seems like it should be super simple to fix. Couldn't we just change the caller to be:

```
    go func() {
        if b.selfHostedEtcd {
            requiredPods = append(requiredPods, "etcd-operator")
            if err := WaitUntilPodsRunning(requiredPods, assetTimeout); err != nil {
                errch <- err
            }
            errch <- etcdutil.Migrate(etcdServiceIP)
        } else {
            errch <- WaitUntilPodsRunning(requiredPods, assetTimeout)
        }
    }()
```